src/irr/video/COpenGLSpecializedShader.cpp

```diff
@@ -114,10 +114,104 @@ COpenGLSpecializedShader::COpenGLSpecializedShader(const video::IVideoDriver* _d
 
     GLchar logbuf[1u<<12]; //4k
     driver->extGlGetProgramInfoLog(m_GLname, sizeof(logbuf), nullptr, logbuf);
-    os::Printer::log(logbuf, ELL_ERROR);
+    if (logbuf[0])
+        os::Printer::log(logbuf, ELL_ERROR);
 
-    // TODO:
-    // what should be interface for setting push_constants (regular uniform on GL backend) for now?
+    m_introspectionData = core::smart_refctd_ptr<asset::CIntrospectionData>(_introspection);
+}
+
+void COpenGLSpecializedShader::setUniformsImitatingPushConstants(const uint8_t* _pcData)
+{
+    if (m_uniformsList.empty())
+        buildUniformsList();
+
+    using gl = COpenGLExtensionHandler;
+    for (const SUniform& u : m_uniformsList)
```

Comment:
how are the uniforms sorted? by location or by offset in pc-range?

Reply:
I haven't try to make them sorted anyhow, but seems like they are in order they appear in uniform block, so location

Reply:
I'd sort them

Reply:
resolved